### PR TITLE
Branding refresh and VerticalListMenu tweaks

### LINKED ISSUE #215: Improve VerticalListMenu to use divs instead of spans
- [x] `.lucid-VerticalListMenu-Item-content-text` should be a `div` instead of a `span`
- [x] we probably don't need [this][foo] anymore
- [x] make sure the new div is 100% width within the flex.

[foo]: https://github.com/appnexus/lucid/blob/v0.2.9/src/components/VerticalListMenu/VerticalListMenu.less#L41

### LINKED ISSUE #233: VerticalListMenu tweaks
- [x] With the work that's being done in #231, we should update `VerticalListMenu` to animate the left border on hover to match that style.
- [x] There are cases where it doesn't make sense to have items that are _expandable_ yet still style their hover as _selectable_. Come up with a good way to make that configurable.

### PR DESCRIPTION
**Pull request by jondlm**
- Fixed #233
- Fixed #215
- Updated `buildHybridComponent` to accept `reducers` as an option so I could override the behavior for the demo page

## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [x] IE11 (Win 7)
  - [x] Edge (Win 10)
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [x] Two core team engineer approvals
- [x] One core team UX approval

### CODE REVIEW

File: src/docs/containers/color-palette.less [hunk 1]
```diff
@@ -88,32 +89,30 @@
...
+		color: contrast(@@name);
+	}
 
 	// Recurse
 	.ColorPalette-loop(@i - 1);
 }
 
-.lucid-ColorPalette {
+.ColorPalette {
+	padding: 15px;
```
Comment (sodiumjoe) [inline src/docs/containers/color-palette.less]: should this be a variable

File: src/docs/containers/color-palette.jsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import React from 'react';
 
 import { lucidClassNames } from '../../util/style-helpers';
 
-const cx = lucidClassNames.bind('&-ColorPalette');
+const cx = lucidClassNames.bind('ColorPalette');
```
Comment (sodiumjoe) [inline src/docs/containers/color-palette.jsx]: why not `lucid-` scope?

File: src/docs/index.jsx [hunk 3]
```diff
@@ -225,11 +242,11 @@ const Component = React.createClass({
 			.value();
 
 		return (
-			<div>
+			<div className='Component'>
 				<h2>{componentName} {privateString} {composesComponents}</h2>
 				<div dangerouslySetInnerHTML={descriptionAsHTML} />
```
Comment (sodiumjoe) [inline src/docs/index.jsx]: ooh dangerous

File: src/docs/index.jsx [hunk 4]
```diff
@@ -352,36 +369,54 @@ const Component = React.createClass({
...
-				{_.map(items, (componentName) => {
+			<VerticalListMenu selectedIndices={[]}>
+				{_.map(_.sortBy(items), (componentName) => {
 					return (
-						<li key={componentName}>
-							<Link to={`/components/${componentName}`}>{componentName}</Link>
-						</li>
+						<Item
+							key={componentName}
+							onSelect={_.partial(this.goToPath, `/components/${componentName}`)}
```
Comment (sodiumjoe) [inline src/docs/index.jsx]: why not just make these `<Router.Link>`s?

File: src/docs/containers/color-palette.jsx (hunk 2 above)
Comment (jondlm) [inline src/docs/containers/color-palette.jsx]: Cause our demo page is equivalent to a consuming application

File: src/docs/index.jsx (hunk 3 above)
Comment (jondlm) [inline src/docs/index.jsx]: Hah, not really

File: src/util/state-management.js [hunk 5]
```diff
@@ -89,18 +89,18 @@ export function safeMerge (objValue, srcValue) {
...
 }
 
-export function buildHybridComponent(baseComponent, opts = {
-	setStateWithNewProps: true, // if true, new props will update state, else prop has priority over existing state
-	replaceEvents: false // if true, function props replace the existing reducers, else they are invoked *after* state reducer returns
-}) {
+export function buildHybridComponent(baseComponent, {
+	setStateWithNewProps = true, // if true, new props will update state, else prop has priority over existing state
+	replaceEvents = false, // if true, function props replace the existing reducers, else they are invoked *after* state reducer returns
+	reducers = _.get(baseComponent, 'definition.statics.reducers', {}),
```
Comment (ogupte) [inline src/util/state-management.js]: nice

File: src/docs/index.jsx (hunk 4 above)
Comment (jondlm) [inline src/docs/index.jsx]: Cause `Link`s render `<a>` tags and I want to navigate when the user selects an item in the `VerticalListMenu`

File: src/util/state-management.js (hunk 5 above)
Comment (jondlm) [inline src/util/state-management.js]: Yup! I wrote a test for it too.

File: src/docs/containers/color-palette.less (hunk 1 above)
Comment (jondlm) [inline src/docs/containers/color-palette.less]: Yeah probably, I use it in lots of places. I've been cutting some corners on the demo page :/

File: src/docs/index.jsx (hunk 4 above)
Comment (ogupte) [inline src/docs/index.jsx]: i think it's bc VerticalListMenu.Item is the thing that should trigger navigation, not just link content of that item.

Comment (sodiumjoe) [inline src/docs/index.jsx]: seems like this is a really common use-case that should interoperate nicely with `Router.Link`

**Pull request merged**